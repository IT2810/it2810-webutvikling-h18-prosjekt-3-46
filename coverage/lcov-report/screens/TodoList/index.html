<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for screens/TodoList</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> screens/TodoList
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">29.79% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>14/47</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">20% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.5% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/24</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">31.82% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>14/44</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="29.79" class="pic low"><div class="chart"><div class="cover-fill" style="width: 29%;"></div><div class="cover-empty" style="width:71%;"></div></div></td>
	<td data-value="29.79" class="pct low">29.79%</td>
	<td data-value="47" class="abs low">14/47</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="20" class="abs low">4/20</td>
	<td data-value="12.5" class="pct low">12.5%</td>
	<td data-value="24" class="abs low">3/24</td>
	<td data-value="31.82" class="pct low">31.82%</td>
	<td data-value="44" class="abs low">14/44</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 19 2018 23:05:13 GMT+0200 (Central European Summer Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
