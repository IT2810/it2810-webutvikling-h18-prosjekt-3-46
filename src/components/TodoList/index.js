import React, { Component } from "react";
import {
    AppRegistry,
    StyleSheet,
    Text,
    View,
    FlatList,
    AsyncStorage,
    TextInput,
    Keyboard,
    Platform,
    TouchableOpacity,
    StatusBar,
    Alert
} from "react-native";
import Swipeout from 'react-native-swipeout';
import Toast, {DURATION} from 'react-native-easy-toast'
import {Button, Icon} from "native-base";

const isAndroid = Platform.OS === "android";
const viewPadding = 0;

export default class TodoList extends Component {

    constructor(props) {
        super(props);

        // Avoid async tasks being run after component is unmounted:
        this._isMounted = false;
    }

    state = {
        tasks: [],
        text: ""
    };

    // React Navigation configuration
    static navigationOptions = {
        title: 'Tasks',

        // Right Header-button.
        headerRight: (
            <Button transparent onPress={() => Alert.alert("Info", "Swipe left on a task in order to remove it.")}>
                <Icon name="ios-information-circle-outline" type="Ionicons"/>
            </Button>
        )
    };


    // LIFECYCLE METHODS

    componentDidMount() {
        this._isMounted = true;

        // Move position of textInput if Keyboard is shown.
        Keyboard.addListener(
            isAndroid ? "keyboardDidShow" : "keyboardWillShow",
            e => this._isMounted && this.setState({ viewPadding: e.endCoordinates.height + viewPadding - 50})
        );

        Keyboard.addListener(
            isAndroid ? "keyboardDidHide" : "keyboardWillHide",
            () => this._isMounted && this.setState({ viewPadding: viewPadding })
        );

        // Set internal list of tasks (in state) if component is mounted. Gets items from AsyncStorage
        Tasks.all(tasks => this._isMounted && this.setState({ tasks: tasks || [] }));

        // Changes the color of the status bar to fit with the displayed content
        this._navListener = this.props.navigation.addListener('didFocus', () => {
            StatusBar.setBarStyle('default');
        });
    }

    componentWillUnmount() {
        this._navListener.remove();
        this._isMounted = false;
    }

    // END LIFECYCLE METHODS


    // Update internal text-state as TextInput is written to.
    changeTextHandler = text => {
        this.setState({ text: text });
    };

    // Handler that is run when the user adds a task.
    addTask = () => {
        let notEmpty = this.state.text.trim().length > 0;

        if (notEmpty) {
            this.setState(
                prevState => {
                    let { tasks, text } = prevState;
                    return {
                        tasks: tasks.concat({ key: tasks.length, text: text }),
                        text: ""
                    };
                },
                () => Tasks.save(this.state.tasks)
            );
        }
    };

    // Handler that is run when the user deletes a task.
    deleteTask = i => {
        this.setState(
            prevState => {
                let tasks = prevState.tasks.slice();

                tasks.splice(i, 1);

                return { tasks: tasks };
            },
            () => Tasks.save(this.state.tasks)
        );
        this.refs.toast.show('Good job!', 2000);
    };

    // Render a single task with corresponding swipeout-component
    renderTask(item, index) {
        let swipeoutBtns = [
            {
                text: "Done",
                backgroundColor: "#5cb85c",
                onPress: () => {
                    this.deleteTask(index)
                }
            }
        ];
        return(
            <View>
                <View style={styles.listItemCont}>
                    <Swipeout right={swipeoutBtns}
                              autoClose={true}
                              style={styles.listSwipe}
                    >
                        <View>
                            <TouchableOpacity>
                                <Text style={styles.listItem}>
                                    {item.text}
                                </Text>
                            </TouchableOpacity>
                        </View>
                    </Swipeout>
                </View>
                <View style={styles.hr} />
            </View>
        )
    }


    render() {
        return (
            <View style={[styles.container, { paddingBottom: this.state.viewPadding }]}>
                <FlatList
                    keyExtractor = { (item, index) => index.toString() }
                    style={styles.list}
                    data={this.state.tasks}
                    renderItem={({ item, index }) => this.renderTask(item, index)}
                />
                <View>
                </View>
                <View style={styles.textContainer}>
                    <TextInput
                        style={styles.textInput}
                        onChangeText={this.changeTextHandler}
                        onSubmitEditing={this.addTask}
                        value={this.state.text}
                        placeholder="Add Tasks"
                        returnKeyType="done"
                        returnKeyLabel="done"
                        underlineColorAndroid='transparent'
                    />
                </View>
                <Toast ref="toast"
                       style={{backgroundColor:"#5cb85c"}}
                       position='center'
                       fadeInDuration={400}
                       fadeOutDuration={900}
                       opacity={1}
                       textStyle={{color:'white', fontSize: 18, padding: 7,
                           paddingLeft: 19, paddingRight: 19}}
                />
            </View>
        );
    }

}

// Retrieve data from AsyncStorage. The list of tasks is stored as a String and is then
// parsed (by splitting) into an Array for setting the internal State.
let Tasks = {
    convertToArrayOfObject(tasks, callback) {
        return callback(
            tasks ? tasks.split("||").map((task, i) => ({ key: i, text: task })) : []
        );
    },
    convertToStringWithSeparators(tasks) {
        return tasks.map(task => task.text).join("||");
    },
    all(callback) {
        return AsyncStorage.getItem("TASKS", (err, tasks) =>
            this.convertToArrayOfObject(tasks, callback)
        );
    },
    save(tasks) {
        AsyncStorage.setItem("TASKS", this.convertToStringWithSeparators(tasks));
    }
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: "center",
        alignItems: "center",
        backgroundColor: '#fff',
        padding: viewPadding,
        paddingTop: 0,
        width: "100%"
    },
    list: {
        width: "100%"
    },
    listItem: {
        padding: 15,
        fontSize: 18,
        paddingTop: 30,
        paddingBottom: 30,
        width: "100%"

    },
    listSwipe: {
        width: "100%",
        backgroundColor: "transparent"
    },
    hr: {
        height: 1,
        backgroundColor: "#EAEAEA"
    },
    listItemCont: {
        flexDirection: "row",
        alignItems: "center",
        justifyContent: "space-between",
        width: "100%"
    },
    textInput: {
        height: 50,
        paddingRight: 15,
        paddingLeft: 15,
        borderColor: "#F2F2F2",
        borderWidth: isAndroid ? 0 : 2,
        width: "100%",
        borderRadius: 15,
        backgroundColor: "white",
        fontSize: 18
    },
    textContainer: {
        backgroundColor: "#EAEAEA",
        width: "100%",
        padding: 10
    }
});

AppRegistry.registerComponent("TodoList", () => TodoList);