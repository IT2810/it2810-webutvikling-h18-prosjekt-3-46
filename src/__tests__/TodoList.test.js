import React from 'react';

import TodoList from '../screens/TodoList/index';
import renderer from 'react-test-renderer';


test('renders correctly', ()=> {
   const tree = renderer.create(<TodoList/>).toJSON();
   expect(tree).toMatchSnapshot();
});
