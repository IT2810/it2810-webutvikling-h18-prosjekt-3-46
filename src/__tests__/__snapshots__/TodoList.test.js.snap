// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
        "padding": 0,
        "paddingTop": 0,
        "width": "100%",
      },
      Object {
        "paddingBottom": undefined,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#EAEAEA",
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Add Tasks"
      returnKeyLabel="done"
      returnKeyType="done"
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#F2F2F2",
          "borderRadius": 15,
          "borderWidth": 2,
          "fontSize": 18,
          "height": 50,
          "paddingLeft": 15,
          "paddingRight": 15,
          "width": "100%",
        }
      }
      value=""
    />
  </View>
</View>
`;
